network/endpoint_windows_test.go

```diff
@@ -504,3 +531,358 @@ func TestNewEndpointImplHnsv2ForIBUnHappyPath(t *testing.T) {
+		t.Fatalf("Failed to create network for infraNIC due to %v", err)
+	}
+
+	// create network for DelegatedNIC
+	delegatedNetwork := &hcn.HostComputeNetwork{
+		Id:   networkID,
+		Name: networkName,
+	}
+	_, err = Hnsv2.CreateNetwork(delegatedNetwork)
+	if err != nil {
+		t.Fatalf("Failed to create network for delegatedNIC due to %v", err)
+	}
+
+	// make sure two networks are created:
+	networks := hnsFake.Cache.GetNetworks()
+	if len(networks) != 2 {
+		t.Fatal("Failed to create two networks for infraNIC and delegatedNIC")
+	}
+
+	// create endpoint for InfraNIC
+	infraEndpointID := "infraEndpoint"
+	infraEndpoint := &hcn.HostComputeEndpoint{
+		Id:                 infraEndpointID,
+		Name:               infraEndpointID,
+		HostComputeNetwork: infraNetworkID,
+	}
+
+	_, err = Hnsv2.CreateEndpoint(infraEndpoint)
+	if err != nil {
+		t.Fatalf("Failed to create endpoint for infraNIC due to %v", err)
+	}
+
+	// create endpoint for delegated NIC
+	deletegatedEndpointID := "delegatedEndpoint"
+	delegatedEndpoint := &hcn.HostComputeEndpoint{
+		Id:                 deletegatedEndpointID,
+		Name:               deletegatedEndpointID,
+		HostComputeNetwork: deletegatedEndpointID,
+		MacAddress:         macAddress,
+	}
+
+	_, err = Hnsv2.CreateEndpoint(delegatedEndpoint)
+	if err != nil {
+		t.Fatalf("Failed to create endpoint for delegatedNIC due to %v", err)
+	}
+
+	// make sure two endpoints are created:
+	endpoints := hnsFake.Cache.GetEndpoints()
+	if len(endpoints) != 2 {
+		t.Fatal("Failed to create two endpoints for infraNIC and delegatedNIC")
+	}
+
+	infraEpInfo := &EndpointInfo{
+		EndpointID:    infraEndpointID,
+		Data:          make(map[string]interface{}),
+		IfName:        "eth0",
+		NICType:       cns.InfraNIC,
+		HNSEndpointID: infraEndpointID,
+		HNSNetworkID:  infraNetworkID,
+	}
+
+	delegatedEpInfo := &EndpointInfo{
+		EndpointID:    deletegatedEndpointID,
+		Data:          make(map[string]interface{}),
+		IfName:        "eth1",
+		NICType:       cns.DelegatedVMNIC,
+		MacAddress:    net.HardwareAddr(macAddress),
+		HNSEndpointID: deletegatedEndpointID,
+		HNSNetworkID:  networkID,
+	}
+
+	// mock DeleteEndpointState() to make sure endpoint and network is deleted from cache
+	// network and endpoint should be deleted from cache for delegatedNIC
+	err = nm.DeleteEndpointState(networkID, delegatedEpInfo)
+	if err != nil {
+		t.Fatalf("Failed to delete endpoint for delegatedNIC state due to %v", err)
+	}
+
+	// endpoint should be deleted from cache for delegatedNIC and network is still there
+	err = nm.DeleteEndpointState(infraNetworkID, infraEpInfo)
+	if err != nil {
+		t.Fatalf("Failed to delete endpoint for delegatedNIC state due to %v", err)
+	}
+
+	// check cache if endpoints are deleted
+	endpoints = hnsFake.Cache.GetEndpoints()
+	if len(endpoints) != 0 {
+		t.Fatalf("Not all endpoints are deleted, the remaining endpoints are %v", endpoints)
+	}
+
+	// check cache if accelnet network is deleted and infra network is still there
```

Comment:
infra network shouldn't be deleted..correct? so len(networks) should be 1

Reply:
also after deletion, the remaining network id is that infra network id

Reply:
solved

Reply:
can check be `len(networks) == 1` ?

Reply:
yes we should check if networks map has one network left